examples/jsm/renderers/common/Background.js

```diff
@@ -49,31 +49,33 @@ class Background extends DataMap {
 			_clearColor.copy( renderer._clearColor );
 			_clearAlpha = renderer._clearAlpha;
 
-			let boxMesh = this.boxMesh;
+			let backgroundMesh = this.backgroundMesh;
 
-			if ( boxMesh === null ) {
+			if ( backgroundMesh === null ) {
 
-				this.boxMeshNode = context( backgroundNode, {
+				this.backgroundMeshNode = context( backgroundNode, {
 					// @TODO: Add Texture2D support using node context
-					getUVNode: () => positionWorldDirection,
+					getUVNode: () => normalWorld,
 					getSamplerLevelNode: () => backgroundBlurriness
 				} ).mul( backgroundIntensity );
 
-				const nodeMaterial = new MeshBasicNodeMaterial();
-				nodeMaterial.colorNode = this.boxMeshNode;
+				let viewProj = modelViewProjection();
+				viewProj = vec4( viewProj.x, viewProj.y, viewProj.w, viewProj.w );
```

Comment:
Maybe just `viewProj.z.assign( viewProj.w )`?

Reply:
If I were to go with the flow I would say that this should return `z`, because the last property accessed would still be `z`, I think that to return the entire vector we should have a function directly on the vector without using `property accessors`, this can confuse the flow of reading, for example I think `viewProj.setZ( viewProj.w )` sounds better in this case.

Reply:
`viewProj.setZ( viewProj.w )` reads much better 👌

Reply:
> If I were to go with the flow I would say that this should return z, because the last property accessed would still be z, I think that to return the entire vector we should have a function directly on the vector without using property accessors, 

I think we discussed this issue a few months ago and agreed that just adding `SplitNode.assign()` will allow to write `vec.x.assign( anything )`.

Reply:
If we split the line, we will see that there is something very strange with `viewProj.z.assign( viewProj.w )`:

```js
const viewProjZ = viewProj.z;

// I think this should return z property not a vec3, so it cannot be inline as suggested
viewProjZ.assign( viewProj.w ) 
```

We don't have the same problem with `node.set*()` because it has no `property accessor`.

Reply:
`viewProjZ` internally is just `new SplitNode( viewProj, 'z' )`, no? I don't see what would be the problem for implementing `assign` for it then...

Reply:
`nove.set*()` also follows the current JS API style.

Reply:
But what I'm suggesting is: 1) more powerful (you can do anything in all combinations, like `vec.xy.assign( anything )`), 2) better reflecting shader code, 3) easier to understand, 4) easier to implement. I can file a PR for it if needed.

Reply:
So only good things?
You can't think of problems that it creates?

Reply:
It is totally possible made `node.setXY()` or any combination using `Proxy`,  I always considered that when I use `node.set*()`.

Reply:
> You can't think of problems that it creates?

I don't really see what problems could it create. It would function similar to `VarNode.assign()`.